cli/src/commands/fix.rs

```diff
@@ -185,39 +191,49 @@ pub(crate) fn cmd_fix(
             }
         }
 
-        // Also fix any new paths that were changed in this commit.
+        // There are four behaviors for the combinations of specifying or omitting the
+        // --all-files flag or file arguments. The flag determines whether we diff the
+        // revision against its parents or just fix the entire tree. In either case, the
+        // file arguments filter that set of files.
```

Comment:
It could be implemented as diffing from the empty tree. I expect the performance difference not to be noticeable.

Reply:
That is more like what `hg fix` does, but I would rather not rework this right now.  The "baserevs" logic for doing the diff in the `include_unchanged_files == false` case is not very complicated, but it's more complicated than what we have here right now.

Reply:
> I expect the performance difference not to be noticeable.

Actually, I take that back. Using `entries_matching` can be a lot slower at Google because it doesn't do the parallel tree walk that `diff_stream()` does. We should fix that, but that shouldn't block this PR.

This is what I had in mind:

```rust
let base_tree = if args.include_unchanged_files {
  MergedTree::resolved(Tree::empty(tx.repo().store().clone(), RepoPathBuf::root()))
} else {
  commit.parent_tree(tx.repo())?
};
let diff_stream = base_tree.diff_stream(&tree, &matcher);
/// as before
```

I don't think it's more complicated than what we have right now, but I'm fine with either form. I think using `entries_matching()` is conceptually simpler because it's more direct. Again, up to you.

Reply:
Diffing against an empty tree seems to use fewer Rust concepts, so I'll go with that.